CMakeLists.txt

```diff
@@ -31,9 +31,10 @@ set(NO_LAPACKE 1)
 endif()
 
 if(BUILD_DEBUG)
-set(CMAKE_BUILD_TYPE Debug)
+	set(CMAKE_BUILD_TYPE Debug)
+	set(OpenBLAS_LIBNAME "${OpenBLAS_LIBNAME}_d")
 else()
-set(CMAKE_BUILD_TYPE Release)
+	set(CMAKE_BUILD_TYPE Release)
 endif()
```

Comment:
More to the point, should this block become something like
if( NOT CMAKE_BUILD_TYPE )
  set( CMAKE_BUILD_TYPE Release)
endif()
if (CMAKE_BUILD_TYPE = Debug)
 set (DEBUG_POSTFIX "_d")
 set(OpenBLAS_LIBNAME "${OpenBLAS_LIBNAME}_d")
else()
 set DEBUG_POSTFIX ""
endif()
which would also consolidate the definitions of DEBUG_POSTFIX as in your other comment ?